amazon/ionbenchmark/benchmark_runner.py

```diff
@@ -112,23 +116,90 @@ def test_fn():
 
     elif match_arg == ['file', 'read', 'load_dump']:
         data_file = benchmark_spec.get_input_file()
-
-        def test_fn():
-            with open(data_file, "rb") as f:
-                return loader_dumper.load(f)
+        format_option = benchmark_spec.get_format()
+        if _format.format_is_ion(format_option):
+            if return_obj:
+                def test_fn():
+                    rtn = []
+                    with open(data_file, "rb") as f:
+                        it = loader_dumper.load(f, parse_eagerly=False)
+                        while True:
+                            try:
+                                rtn.append(next(it))
+                            except StopIteration:
+                                break
+                    return rtn
+            else:
+                def test_fn():
+                    with open(data_file, 'br') as f:
+                        for v in loader_dumper.load(f, parse_eagerly=False):
+                            pass
+        elif _format.format_is_json(format_option):
+            if return_obj:
+                def test_fn():
+                    rtn = []
+                    with open(data_file, 'r') as f:
+                        for v in loader_dumper.load(f):
+                            rtn.append(v)
+                    return rtn
+            else:
+                def test_fn():
+                    with open(data_file, 'r') as f:
+                        for v in loader_dumper.load(f):
+                            pass
+        elif _format.format_is_cbor(format_option):
+            if return_obj:
+                def test_fn():
+                    rtn = []
+                    with open(data_file, 'br') as f:
+                        for v in loader_dumper.load(f):
+                            rtn.append(v)
+                    return rtn
+            else:
+                def test_fn():
+                    with open(data_file, 'br') as f:
+                        for v in loader_dumper.load(f):
+                            pass
+        elif _format.format_is_protobuf(format_option):
+            # Refer to https://github.com/amazon-ion/ion-python/issues/326
+            raise NotImplementedError("Benchmarking Protocol Buffer multiple top level object use case may not "
+                                      "support yet.")
 
     elif match_arg == ['file', 'write', 'load_dump']:
+        # This method should return a list that holds all top_level values.
```

Comment:
Why? Why not return a generator?

Reply:
Okay yeah, we can reuse the `loader.load` for that. Working on this.